### PR TITLE
Fix compatiblity tests failures after removing legacy error codes.

### PR DESCRIPTION
**Pull request by kamil-da**
CHANGELOG_BEGIN
CHANGELOG_END

run-full-compat: true

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compatibility/bazel_tools/testing.bzl [hunk 1]
```diff
@@ -1064,10 +1094,11 @@ def sdk_platform_test(sdk_version, platform_version):
...
     # Error codes are enabled by default after 1.18.0-snapshot.20211117.8399.0.a05a40ae.
     # Before this SDK version, ledger-api-test-tool cannot correctly assert the self-service error codes.
-    # For this reason, all platforms newer than 1.18.0-snapshot.20211117.8399.0.a05a40ae will run against
-    # old ledger-api-test-tools in compatibility mode (i.e. `--use-pre-1.18-error-codes`)
+    # Turning on the compatibility mode with `--use-pre-1.18-error-codes` is available up to 2.0.0-snapshot.20220127.9042.0.4038d0a7 (inclusive).
+    # For this reason, all platforms newer than 1.18.0-snapshot.20211117.8399.0.a05a40ae up to 2.0.0-snapshot.20220127.9042.0.4038d0a7 (inclusive)
+    # will run against old ledger-api-test-tools in compatibility mode (i.e. `--use-pre-1.18-error-codes`)
     error_codes_version_enabled_by_default = "1.18.0-snapshot.20211117.8399.0.a05a40ae"
-    if versions.is_at_most(error_codes_version_enabled_by_default, sdk_version):
+    if versions.is_at_most(error_codes_version_enabled_by_default, platform_version) and versions.is_at_least(before_removing_legacy_error_codes, platform_version):
```
Comment (cocreature) [inline compatibility/bazel_tools/testing.bzl:1101]: do we not need to guard on the sdk version to figure out if ledger API test tool is testing against a new or an old sdk version?

Comment (kamil-da) [reply compatibility/bazel_tools/testing.bzl:1101]: As far as I understand this part we just want to say that ledgers in a specific range should run in error codes compatibility mode. The `test-tool` adjustments should be done in `exclusions` section.

Comment (mziolekda) [reply compatibility/bazel_tools/testing.bzl:1101]: Hmmm
```if versions.is_at_most(error_codes_version_enabled_by_default, platform_version) and versions.is_at_least(before_removing_legacy_error_codes, platform_version)```

That looks like `v <=1.18.0.snapshot && v >= 2.0.0.snapshot` so an empty set. Am I missing something?

Comment (kamil-da) [reply compatibility/bazel_tools/testing.bzl:1101]: I've double or triple check that. The naming is really confusing here but I think I got it right :)

**Review by cocreature — APPROVED** [review]

File: compatibility/bazel_tools/testing.bzl (hunk 1 above)
Comment (mziolekda) [reply compatibility/bazel_tools/testing.bzl:1101]: Yes you are right, the method name is the wrong way round! :facepalm:

Comment (cocreature) [reply compatibility/bazel_tools/testing.bzl:1101]: yeah it’s weird, we could change it but this is the order the upstream bazel ilbraries use so keeping it consistent seemed potentially less confusing.

Comment (mziolekda) [reply compatibility/bazel_tools/testing.bzl:1101]: So the is_at_most/is_at_least methods do what their name suggest, not what their help is saying
But that means that the code is selecting an empty set

**Pull request merged**